MaxText/layers/moe.py

```diff
@@ -782,13 +780,12 @@ def dense_matmul(self, inputs, gate_logits, pre_bias_logits, w0_kernel, w1_kerne
 
     cp, sub_seq = self.get_context_partition_and_sub_seq(seq_len)
 
-    # TODO @jacobplatin: allow Llama4 MoE to handle capacity factor > 0
     if self.config.capacity_factor > 0:
       # token dropping if needed
       if self.config.model_call_mode != "inference":
+        weights = self.reshape_and_update_weights(top_k_weights, top_k_indices)
```

Comment:
We saw an exhausted resource issue in the test. If this still happens in tests, could you remove `weights` back, and then we could follow it up with a new PR. Thank you!